### PR TITLE
Docker V0 and V1 store build output in temp file

### PR DESCRIPTION
**Pull request by jcfiorenzano**
**Task name**:  Docker

**Description**: the build command is going to store the output of docker in a temp file and is going to store its path in an output variable

**Added unit tests:** N

**Attached related issue:** https://github.com/microsoft/azure-pipelines-tasks/issues/14537

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/DockerV0/utils.ts [hunk 1]
```diff
@@ -82,6 +85,28 @@ export function getImageMappings(connection: ContainerConnection, imageNames: st
     return sourceToTargetMapping;
 }
 
+
+function getTaskOutputDir(command: string): string {
```
Comment (jcfiorenzano) [inline Tasks/DockerV0/utils.ts:89]: This section was copied from the Utils file from DockerV2 I would like to know if there is a process to share functions between the different tasks.

File: Tasks/DockerV1/containerbuild.ts [hunk 2]
```diff
@@ -66,5 +66,11 @@ export function run(connection: ContainerConnection): any {
         context = tl.getPathInput("buildContext");
     }
     command.arg(context);
+
+    command.on("stdout", data => {
+        let taskOutputPath = utils.writeTaskOutput("build", data);
+        tl.setVariable("DockerOutputPath", taskOutputPath);
```
Comment (jcfiorenzano) [inline Tasks/DockerV1/containerbuild.ts:72]: I would like to know if there is a way for testing this. I tried using `process.env` but the variable was not set there even though it was set correctly in the code ( if you print `tf.stdoutput` you can see that the variable was set correctly). I also didn't see an example of a test for the regular output variable `DockerOutput`

File: Tasks/DockerV0/containerbuild.ts [hunk 3]
```diff
@@ -63,5 +63,11 @@ export function run(connection: ContainerConnection): any {
         context = tl.getPathInput("context");
     }
     command.arg(context);
+
+    command.on("stdout", data => {
+        let taskOutputPath = utils.writeTaskOutput("build", data);
```
Comment (ajinkya599) [inline Tasks/DockerV0/containerbuild.ts:68]: This should happen at the end (`stdout` event can happen in chunks).
On `stdout` event, append to a local variable and write to file and set the variable after command execution completes.

Comment (ajinkya599) [reply Tasks/DockerV0/containerbuild.ts:68]: `Suggestion`:
```
return connection.execCommand(command).then(() => {
  // write to file and output variable.
});```

File: Tasks/DockerV1/containerbuild.ts [hunk 4]
```diff
@@ -66,5 +66,11 @@ export function run(connection: ContainerConnection): any {
         context = tl.getPathInput("buildContext");
     }
     command.arg(context);
+
+    command.on("stdout", data => {
```
Comment (ajinkya599) [inline Tasks/DockerV1/containerbuild.ts:70]: This should happen at the end (`stdout` event can happen in chunks).
On `stdout` event, append to a local variable and write to file and set the variable after command execution completes.

File: Tasks/DockerV1/containerbuild.ts (hunk 2 above)
Comment (ajinkya599) [reply Tasks/DockerV1/containerbuild.ts:72]: I don't know a place where output variable is being tested.
The [documentation](https://docs.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#setvariable-initialize-or-modify-the-value-of-a-variable) says the variable is available as env variable to the tasks that follow.
Maybe test that the command to set the variable is present in `stdout`?

Also, can you pls test it out by actually running in a pipeline and adding a script to output this to the logs?

**Comment by jcfiorenzano** [comment]
Hi @ajinkya599 this is the result of this PR running in a pipeline:

Variable set:
image

File content:
```
Sending build context to Docker daemon  66.05kB

Step 1/1 : FROM ubuntu:16.04
 ---> aefd7f02ae24
Successfully built aefd7f02ae24
Successfully tagged test-docker:31
SECURITY WARNING: You are building a Docker image from Windows against a non-Windows Docker host. All files and directories added to build context will have '-rwxr-xr-x' permissions. It is recommended to double check and reset permissions for sensitive files and directories.

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
```

**Review by ajinkya599 — APPROVED** [review]
🚀

**Pull request merged**